sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2ScanExecBase.scala

```diff
@@ -168,6 +168,18 @@ trait DataSourceV2ScanExecBase extends LeafExecNode {
     }
   }
 
+  protected def postDriverMetrics(): Unit = {
+    val driveSQLMetrics = scan.reportCustomDriverMetrics().map( customTaskMetric => {
+      val metric = metrics(customTaskMetric.name())
```

Comment:
hmm what if the driver metric is not in `metrics`? this will throw `NoSuchElementException`?

Reply:
Yea, I think so. Considering this is implemented by the developers for DS v2 data sources, I don't think this would happen. Otherwise it is caught during development.